### PR TITLE
Ejercicios Cine y Filtros Unai Castro

### PR DESCRIPTION
**Pull request by UnaiCastro**
This pull request includes significant updates to two JavaScript files, `Cinema.js` and `actividad.js`. The changes in `Cinema.js` focus on setting up a cinema seating arrangement and implementing a function to suggest available seats. The changes in `actividad.js` involve adding multiple image processing functions, such as converting images to different color scales and merging images.

### Changes in `Cinema.js`:

* Added a function to initialize a matrix of cinema seats with unique IDs and occupancy status.
* Implemented a `suggest` function to recommend available seats based on the number of seats requested.

### Changes in `actividad.js`:

* Added various image processing functions:
  - `redConverter`, `greenConverter`, `blueConverter` to convert images to red, green, and blue scales respectively.
  - `greyConverter` to convert images to grayscale.
  - `blackAndWhiteConverter` to convert images to black and white.
  - `scaleDown` to reduce the image size by removing even-indexed rows and columns.
  - `dimBrightness` to reduce the image brightness by a given factor.
  - `invertColors` to invert the colors of the image.
  - `merge` to blend two images together using specified alpha values.
* Added a test program to execute the image processing functions based on a given option.

### CODE REVIEW

File: Tema_2/Unai_Castro_MartínezDeMendibil/Cine/Cinema.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    for (let i = butacas.length-1; i >=0; i--) {
+        let fila = butacas[i];
+        let contador = 0;
+        let ids = new Set();
+        for (let j = 0; j < fila.length; j++) {
+            if (fila[j].estado === false) {
+                contador++;
+                ids.add(fila[j].id);
+                if (contador === n) {
+                    return ids;
```
Comment (UnirCs) [inline Tema_2/Unai_Castro_MartínezDeMendibil/Cine/Cinema.js:56]: la invariante!!

**Review by UnirCs — CHANGES_REQUESTED** [review]
Salvo por la violacion que haces de los invariantes, todo correcto. Refactoriza el código para no mentir acerca de cuándo terminan o no los bucles :)

**Comment by UnaiCastro** [comment]
He pusheado mi ultimo commit con los cambios que he hecho quitando la invariante. No se si teniendo ya el pull request hecho de antes no hace falta volver a hacer pull pero no me deja hacer pull request.

Espero que la solución se pueda ver y este bien porque mira que recalcaste en la clase que no usemos la invariante y he caído...

Un saludo!

**Review by UnirCs — APPROVED** [review]
Así está genial Unai, al tener la PR abierta automáticamente se cambia el código aquí también.

**Pull request merged**